src/Tvheadend.cpp

```diff
@@ -367,6 +367,34 @@ PVR_ERROR CTvheadend::GetChannels(bool radio, kodi::addon::PVRChannelsResultSet&
   return PVR_ERROR_NO_ERROR;
 }
 
+PVR_ERROR CTvheadend::GetChannelStreamProperties(const kodi::addon::PVRChannel& channel, std::vector<kodi::addon::PVRStreamProperty>& properties)
+{
+  CLockObject lock(m_mutex);
+
+  if (!Settings::GetInstance().GetStreamingHTTP())
+    return PVR_ERROR_NO_ERROR;
+
+  auto it = m_channels.find(channel.GetUniqueId());
+  if (it == m_channels.end())
+    return PVR_ERROR_FAILED;
+
+  std::string url = "http://";
```

Comment:
std::string url = Settings::GetInstance().GetUseHTTPS() ? "https://" : "http://";

Reply:
video streaming via https is nonsense. in particular in a home network

Reply:
Why is https nonsense? Why is this limited to home network?

Reply:
> Why is this limited to home network?

do you remember what the 'p' stands for in pvr?

> Why is https nonsense?

it introduces significant overhead

Reply:
> do you remember what the 'p' stands for in pvr?

 Sure, but why does "personal" imply something technical related to network protocols? Personally I sometimes use a HTSP server located at the other end of europe and it works very well.

Reply:
if you think https is anything of value, feel free to implement later. I won't do this nonsense

Reply:
I do not need http streaming.

Reply:
@FernetMenta nonsense or not, there's an option to use HTTPS and breaking that contract with the user would be pretty bad.

Reply:
When I think about this some more, I think the reason for the HTTPS option is so people don't have to send their username/password in plain text back and forth. Even if we'd skip HTTPS for streaming because it's heavy, we'd still leak the username and password.

The sanest solution would be to implement Digest authentication (https://github.com/kodi-pvr/pvr.hts/issues/386).

Reply:
https isn't even implemented properly here because there is no method to import the server's certificate. makes https pretty useless here. better not having it at all than having it implemented the way it is.

> The sanest solution would be to implement Digest authentication

feel free to do so. I don't see any need for this because I consider it nonsense to expose a tvh server to the public internet. if I want to connect to my tvh backend from remote, I use vpn.

Reply:
There's no need to import any certificate, most people nowadays (I assume) use legitimate certificates and not self-signed.

Anyway, copying the snippet @ksooo started this thread with is enough, we don't need to get philosophical over this feature right now.

Reply:
Who would pay 80€ per year for a trusted certificate? I bet you don't know a single person doing this for their tvh server. I won't add this incomplete https nonsens. If this stops you from merging this, let me know.

Reply:
Who would pay 80€ per year for a trusted certificate? I bet you don't know a single person doing this for their tvh server. I won't add this incomplete https nonsens. If this stops you from merging this, let me know.

Reply:
Let's Encrypt gives anyone free certificates nowadays. I don't personally know anyone doing this, but obviously some do since the HTTPS option was added in the first place.

I'm not sure why you can't just change it to one-liner @ksooo gave you?

Reply:
https://letsencrypt.org
No need to pay for trusted certs these days.

Reply:
> I'm not sure why you can't just change it to one-liner

because I don't do nonsense. this https code in this repo is crap and I won't add to this. why don't you add this line instead of forcing others to do this.

Reply:
If you only could stay polite.- "crap", "nonsense", ... this all does not help a bit, but upsets people, at least those who have spent their time implementing this.